File: backend/src/packaging/service.js
import { BaseService } from "../services/BaseService.js";
import { build } from "./packaging.js";

// PackagingService is the service-shaped adapter over `build`. The factory
// stays as the implementation — the service wrapper gives us an injection
// seam for consumers that prefer a method receiver (and keeps the
// cross-stack service-symmetry story intact).
export class PackagingService extends BaseService {
  constructor() {
    super({ entityName: "packaging" });
  }

  build(size, mode) {
    return build(size, mode);
  }
}

File: store-service/internal/packaging/service.go
package packaging

import "duckstore/store-service/internal/service"

// PackagingService is the service-shaped adapter over Build. The underlying
// pure function stays exported (for internal reuse) — the service just gives
// us an injection seam for consumers that prefer a method receiver.
type PackagingService struct {
	service.BaseService
}

func NewService() *PackagingService {
	return &PackagingService{BaseService: service.New("packaging")}
}

func (s *PackagingService) Build(size Size, mode ShippingMode) (Package, error) {
	return Build(size, mode)
}
